hw_2.1/Trie/Trie/Trie.cs

```diff
@@ -0,0 +1,157 @@
+﻿namespace Trees.Trie
+{
+    // Trie data structure realization
+    public class Trie
+    {
+        private TrieNode head;
+        private int size;
+
+        private class TrieNode
+        {
+            public char value;
+            public int wordsWithSuchPrefix;
+            public bool isTerminated;
+            public Dictionary<char, TrieNode> children;
```

Comment:
```suggestion
            public char Value { get; };
            public int WordsWithSuchPrefix { get; set; };
            public bool IsTerminated { get; set; };
            public Dictionary<char, TrieNode> Children { get; };
```
В C# принято поля использовать для скрытой информации. А публичными делать свойства, так как для них более гибко можно настроить права доступа ([read here](https://learn.microsoft.com/ru-ru/dotnet/csharp/programming-guide/classes-and-structs/fields)) .